src/main/scala/com/fulcrumgenomics/fastq/DemuxFastqs.scala

```diff
@@ -556,17 +572,40 @@ private[fastq] case class SampleInfo(sample: Sample, isUnmatched: Boolean = fals
 private[fastq] object FastqDemultiplexer {
 
   /** Stores the minimal information for a single template read. */
-  case class DemuxRecord(name: String, bases: String, quals: String, molecularBarcode: Option[String],
+  final case class DemuxRecord(name: String, bases: String, quals: String, molecularBarcode: Option[String],
                          sampleBarcode: Option[String], readNumber: Int, pairedEnd: Boolean, comment: Option[String],
-                         originalBases: Option[String] = None, originalQuals: Option[String] = None)
+                         originalBases: Option[String] = None, originalQuals: Option[String] = None) {
+
+    /** A function to mask bases that have a quality score less than or equal to a specified threshold
+      * @param threshold The threshold for masking bases, inclusive. Bases with a quality score <= threshold will be masked
+      * @param qualityEncoding The encoding used for quality scores in the Fastq file.
+      * @return a new DemuxRecord with updated bases
+      */
+    def maskLowQualityBases(threshold: Int, qualityEncoding: QualityEncoding): DemuxRecord = {
+      val bases = this.bases.zip(quals).map { case (base, qual) =>
+        if (qualityEncoding.toStandardNumeric(ch = qual) <= threshold) "N" else base
+      }
+    this.copy(bases = bases.mkString)
```

Comment:
It would be more efficient go translate the other way and ask "what character is `threshold` in the given encoding, then compare to that.

Reply:
I think too if the assumption is that the majority of reads would not have bases below threshold it would be faster to first check, e.g. 

```scala
val thresh = qualityEncoding.toAlpha(threshold)
if (this.quals.exists(_ < = thresh)) {
  // do the masking and return copy(bases=newBases)
}
else {
  this
}
```

Reply:
Also, when masking it would likely be much more efficient to e.g.:

```scala
val bases = this.bases.getBytes
forloop (from=0, until=quals.length) { i => 
  if (quals.charAt(i) <= thresh) bases(i) = 'N'
}
new String(bases)
```

Reply:
1. Agree with @tfenne .  
2. `qualityEncoding.toAlpha(threshold)` could be computed once and stored somewhere so we don't have to do it each time.  Perhaps in the main class and pass the ascii value.
3.  This should just return `this` if the quality threshold is not defined or zero or below.  That would be much faster when using the default value.